### ISSUE TITLE
[BUG] Filters and sorting order not working on page reload

### ISSUE THREAD

**Issue by freb97** [shopware/frontends#687]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When filtering products with the `useListing` composable, the property IDs, manufacturer IDs, min/max-price, etc. are added to the current page URL and the filters get applied correctly, only displaying the products matching the current filters. Same thing for the sorting, it works fine when not leaving the page and choosing a sorting order.

When reloading the page after choosing a filter or sorting order, the product listing should still only show matching products for these filters and sorting. Currently, the listing just resets to the original order and displays all products, not only the matching ones.

This faulty behaviour can also be reproduced on the Vercel demo.

### Expected Behavior

The product filters and sorting should work when applied, not only when set initially but also when reloading the page.

### Steps To Reproduce

1. Open a listing page in the shopware frontends demo store (e.g. https://frontends-demo.vercel.app/Cooking-pleasure-Provence/Drink/)
2. Filter products by price, manufacter or any other filtering method or choose a sorting order
3. Reload the page
4. Products now lose their filters and sorting

**Comment by BrocksiNet** [comment]
I can confirm that this is a bug in production. :warning: 
On the search page, this bug does not occur also not in production (tested [here](https://frontends-demo-git-feat-filter-count-8b03c7-shopware-frontends.vercel.app/search?search=sal&order=topseller)).
When I do the same with my local instance it is working correctly. 
Also when I look at what the server is rendering.

Somehow I think [there](https://github.com/shopware/frontends/blob/main/templates/vue-demo-store/components/FrontendNavigationPage.vue#L16) could be a problem.

**Comment by freb97** [comment]
@BrocksiNet The idea is that the useAsyncData identifier string `"cmsNavigation" + props.navigationId,` could retrieve the cached products ignoring `route.query` as it is not added to the identifier, right?

I've tried adding the queried IDs to the identifier so that it differs when a URL with pre-selected filters is specified, but this doesn't seem to fix the problem so far. The page initially loads correctly and with the filters applied, but after 1-2 seconds the product list suddenly shows all products, regardless of filtering...

I am not sure if this is a mistake on our end, as we have modified the listing page to our customers design specifications, but i thought maybe this information could be useful for further debugging.
